miniapps/meshing/minimal-surface.cpp

```diff
@@ -0,0 +1,1302 @@
+constexpr char vishost[] = "localhost";
+
+// Context/Options for the solver
+struct Opt
+{
+   int sz, id;
+   int pb = 0;
+   int nx = 6;
+   int ny = 6;
+   int order = 3;
+   int refine = 2;
+   int niters = 8;
+   int surface = 5;
+   bool pa = true;
+   bool vis = true;
+   bool amr = false;
+   bool wait = false;
+   bool print = false;
+   bool radial = false;
+   bool by_vdim = false;
+   bool snapshot = false;
+   bool vis_mesh = false;
+   double tau = 1.0;
+   double lambda = 0.1;
+   double amr_threshold = 0.6;
+   const char *keys = "gAm";
+   const char *device_config = "cpu";
+   const char *mesh_file = "../../data/mobius-strip.mesh";
+   void (*Tptr)(const Vector&, Vector&) = nullptr;
+};
+
+class Surface: public Mesh
+{
+protected:
+   Opt &opt;
+   Mesh *mesh;
+   Array<int> bc;
+   H1_FECollection *fec;
+   FiniteElementSpace *fes;
+public:
+   // Reading from mesh file
+   Surface(Opt &opt, const char *file): Mesh(file, true), opt(opt) { }
+
+   // Generate 2D empty surface mesh
+   Surface(Opt &opt, bool): Mesh(), opt(opt) { }
+
+   // Generate 2D quad surface mesh
+   Surface(Opt &opt): Mesh(opt.nx, opt.ny, QUAD, true), opt(opt) { }
+
+   // Generate 2D generic surface mesh
+   Surface(Opt &opt, int nv, int ne, int nbe):
+      Mesh(DIM, nv, ne, nbe, SDIM), opt(opt) { }
+
+   void Create()
+   {
+      if (opt.surface > 0)
+      {
+         Prefix();
+         Transform();
+      }
+      Postfix();
+      Refine();
+      Snap();
+      fec = new H1_FECollection(opt.order, DIM);
+      if (opt.amr) { EnsureNCMesh(); }
+      mesh = new Mesh(*this, true);
+      fes = new FiniteElementSpace(mesh, fec, opt.by_vdim ? 1 : SDIM);
+      BoundaryConditions();
+   }
+
+   int Solve()
+   {
+      // Initialize GLVis server if 'visualization' is set
+      if (opt.vis) { opt.vis = glvis.open(vishost, visport) == 0; }
+      // Send to GLVis the first mesh
+      if (opt.vis) { Visualize(opt, mesh, GLVIZ_W, GLVIZ_H); }
+      // Create and launch the surface solver
+      if (opt.by_vdim)
+      {
+         ByVDim(*this, opt).Solve();
+      }
+      else
+      {
+         ByNodes(*this, opt).Solve();
+      }
+      if (opt.vis && opt.snapshot)
+      {
+         opt.keys = "Sq";
+         Visualize(opt, mesh, mesh->GetNodes());
+      }
+      return 0;
+   }
+
+   ~Surface()
+   {
+      if (opt.vis) { glvis.close(); }
+      delete mesh; delete fec; delete fes;
+   }
+
+   virtual void Prefix()
+   {
+      SetCurvature(opt.order, false, SDIM, Ordering::byNODES);
+   }
+
+   virtual void Transform() { if (opt.Tptr) { Mesh::Transform(opt.Tptr); } }
+
+   virtual void Postfix()
+   {
+      SetCurvature(opt.order, false, SDIM, Ordering::byNODES);
+   }
+
+   virtual void Refine()
+   { for (int l = 0; l < opt.refine; l++) { UniformRefinement(); } }
+
+   virtual void Snap()
+   {
+      GridFunction &nodes = *GetNodes();
+      for (int i = 0; i < nodes.Size(); i++)
+      { if (std::abs(nodes(i)) < EPS) { nodes(i) = 0.0; } }
```

Comment:
is this example conforming to MFEM coding conventions? (i.e. by running `astyle`)

Reply:
The `make style` runs fine; but I agree, I could skip lines there.

Reply:
I added new lines with for-loop statements.

Reply:
`done`